hudi-metastore/src/main/java/org/apache/hudi/metastore/HoodieMetastoreServer.java

```diff
@@ -0,0 +1,123 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.metastore;
+
+import org.apache.hudi.exception.HoodieIOException;
+import org.apache.hudi.metastore.service.HoodieMetastoreService;
+import org.apache.hudi.metastore.service.HoodieMetaStoreProxyHandler;
+import org.apache.hudi.metastore.service.PartitionService;
+import org.apache.hudi.metastore.service.SnapshotService;
+import org.apache.hudi.metastore.service.TableService;
+import org.apache.hudi.metastore.service.TimelineService;
+import org.apache.hudi.metastore.store.RelationDBBasedStore;
+import org.apache.hudi.metastore.store.MetadataStore;
+import org.apache.hudi.metastore.thrift.MetaStoreException;
+import org.apache.hudi.metastore.thrift.ThriftHoodieMetastore;
+import org.apache.hudi.metastore.util.TServerSocketWrapper;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.apache.thrift.server.TServer;
+import org.apache.thrift.server.TThreadPoolServer;
+import org.apache.thrift.transport.TServerTransport;
+
+import java.lang.reflect.Proxy;
+
+public class HoodieMetastoreServer {
+
+  private static final Logger LOG = LogManager.getLogger(HoodieMetastoreServer.class);
+
+  private static volatile TServer server;
+  private static volatile Thread serverThread;
+  private static volatile MetadataStore metadataStore;
+  private static volatile HoodieMetastoreService metastoreService;
+
+  public static void main(String[] args) {
+    startServer();
+  }
+
+  public static void startServer() {
+    try {
+      if (server != null) {
+        return;
+      }
+      metadataStore = new RelationDBBasedStore();
+      // service
+      TableService tableService = new TableService(metadataStore);
+      PartitionService partitionService = new PartitionService(metadataStore);
+      TimelineService timelineService = new TimelineService(metadataStore);
+      SnapshotService snapshotService = new SnapshotService(metadataStore);
+      HoodieMetastoreService hoodieMetastoreService = new HoodieMetastoreService(tableService,
+          partitionService, timelineService, snapshotService);
+      HoodieMetaStoreProxyHandler proxyHandler = new HoodieMetaStoreProxyHandler(hoodieMetastoreService);
+
+      // start a thrift server
+      ThriftHoodieMetastore.Iface proxy = (ThriftHoodieMetastore.Iface) Proxy
+          .newProxyInstance(HoodieMetaStoreProxyHandler.class.getClassLoader(),
+              new Class[]{ThriftHoodieMetastore.Iface.class}, proxyHandler);
+      ThriftHoodieMetastore.Processor processor = new ThriftHoodieMetastore.Processor(proxy);
+      TServerTransport serverTransport = new TServerSocketWrapper(9090);
```

Comment:
Let's make the port configurable

Reply:
We need a. jcommander to do it, will fix later